lib/improver/generate_ancillaries/generate_topographic_zone_weights.py

```diff
@@ -205,6 +205,8 @@ def process(self, orography, thresholds_dict, landmask=None):
                     threshold_units, sea_points_included=sea_points_included))
             topographic_zone_cubes.append(topographic_zone_cube)
         topographic_zone_weights = topographic_zone_cubes.concatenate_cube()
+        topographic_zone_weights.data = (
+            topographic_zone_weights.data.astype(np.float32))
```

Comment:
`_make_mask_cube` returns an int32 cube of boolean(?) values.  Why does `topographic_zone_weights` need to be float32?  Is there some interpolation happening or is this a standard we could reconsider?

Reply:
If you look down to line 250 you will see that this topographic_zone_weights.data is partially replaced with other data. The weights, calculated using self.calculate_weights, are not integer values, so the data array type needs to be float32. However, the _make_mask_cube should return an integer as it is meant to be creating a binary mask. Hence the various changes.

Reply:
OK